plugins/inputs/nats_consumer/nats_consumer.go

```diff
@@ -142,6 +145,33 @@ func (n *natsConsumer) Start(acc telegraf.Accumulator) error {
 
 			n.subs = append(n.subs, sub)
 		}
+
+		if len(n.JsSubjects) > 0 {
+			var connErr error
+			n.jsConn, connErr = n.conn.JetStream(nats.PublishAsyncMaxPending(256))
+			if connErr != nil {
+				return connErr
+			}
+
+			if n.jsConn == nil {
```

Comment:
Shouldn't this be checking if `n.jsConn` isn't nil before continuing? Wouldn't `n.jsConn.QueueSubscribe` fail if it **is** nill?

```suggestion
			if n.jsConn != nil {
```

Reply:
you are right, thank you. have changed it